internal/runners/ppm/convert.go

```diff
@@ -0,0 +1,191 @@
+package ppm
+
+import (
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/failures"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/textutils"
+	"github.com/skratchdot/open-golang/open"
+)
+
+// convertAnswerCreate is the answer that the user can choose if they accept to create a virtual environment.  It is re-used at several places.
+var convertAnswerCreate string = locale.Tl("ppm_convert_answer_create", "Create Virtual Runtime Environment")
+
+// analyticsEventFunc is used to send analytics event
+type analyticsEventFunc func(string, string, string)
+
+// surveySelectFunc displays a menu with options that the user can select
+type surveySelectFunc func(message string, choices []string, defaultResponse string) (string, *failures.Failure)
+
+const (
+	askedWhy          string = "asked why"
+	seenStateToolInfo        = "visited state tool info"
+	seenPlatformInfo         = "visited platform info"
+	notConvinced             = "still wants ppm"
+)
+
+type conversionFlow struct {
+	survey    surveySelectFunc
+	out       output.Outputer
+	openURI   func(string) error
+	eventFunc analyticsEventFunc
+}
+
+// StartConversionFlowIfNecessary checks if the user is in a project directory.
+// If not, they are asked to create a project, and (in a wizard-kind-of way) informed about the consequences.
+func (p *Ppm) StartConversionFlowIfNecessary() error {
+	// start conversion flow only if we cannot find a project file
+	if p.project == nil {
+		cf := newConversionFlow(p.prompt.Select, p.out, open.Run, analytics.EventWithLabel)
```

Comment:
Please remove the unit tests for this and simplify the code. If we really want to test this we can write integration tests, but for now let's just focus on landing this story.

If I got to write the `state deploy` code again I would dump a lot of the logic around unit tests and simplify great portions. The pain that code has caused me is far greater than that the unit tests actually solved any problem we were having.

Reply:
I am not quite convinced, as I believe that the unit-test could be very helpful here, as it can test things that are very difficult or impossible to test with integration tests, but I'll remove it.  It certainly makes the code a little simpler.

Reply:
What wouldn't you be able to test with integration tests? Sounds like a gap in our integration testing framework that we should fill.

Personally my new standard for unit tests is they should be achievable with minimal disruption to your architecture. ie. if you are architecting your code in a very specific way JUST to facilitate unit tests at the cost of readability and maintainability then that code better be worth making the effort for.

Reply:
Well, I was thinking about the google analytics events.  In the unit tests, you can nicely check that they were triggered for each code path, and also check their payload.  It is not *impossible* to intercept these events, but probably not very feasible. But I understand your argument that it maybe is not worth the effort: After all state tool logic is never so complicated that we **have** to do unit tests...  

State Tool complexity comes more from the rather wide spectrum of functionality, and that can be managed with integration tests.